src/battle_anim_fire.c

```diff
@@ -959,94 +980,123 @@ static u16 GetEruptionLaunchRockInitialYPos(u8 spriteId)
     return y;
 }
 
-static void InitEruptionLaunchRockCoordData(struct Sprite *sprite, s16 x, s16 y)
+static void InitEruptionLaunchRockCoordData(struct Sprite *sprite, s16 speedX, s16 speedY)
 {
-    sprite->data[0] = 0;
-    sprite->data[1] = 0;
-    sprite->data[2] = (u16)sprite->x * 8;
-    sprite->data[3] = (u16)sprite->y * 8;
-    sprite->data[4] = x * 8;
-    sprite->data[5] = y * 8;
+    sprite->sSpeedDelay = 0;
+    sprite->sLaunchStage = 0;
+    sprite->sX = (u16)sprite->x * 8;
```

Comment:
```suggestion
    sprite->sX = sprite->x << 3;
```
You can remove the cast by doing a shift by 3